src/miner.cpp

```diff
@@ -1616,21 +1637,8 @@ void static BitcoinMiner()
                         eq.digitK(0);
                         ehSolverRuns.increment();
 
-                        // Convert solution indices to byte array (decompress) and pass it to validBlock method.
-                        for (size_t s = 0; s < eq.nsols; s++) {
-                            LogPrint("pow", "Checking solution %d\n", s+1);
-                            std::vector<eh_index> index_vector(PROOFSIZE);
-                            for (size_t i = 0; i < PROOFSIZE; i++) {
-                                index_vector[i] = eq.sols[s][i];
-                            }
-                            std::vector<unsigned char> sol_char = GetMinimalFromIndices(index_vector, DIGITBITS);
-
-                            if (validBlock(sol_char)) {
-                                // If we find a POW solution, do not try other solutions
-                                // because they become invalid as we created a new block in blockchain.
-                                break;
-                            }
-                        }
+                        check_tromp_solution(eq, validBlock);
```

Comment:
May be here we need:
```
if (check_tromp_solution(eq, validBlock))
    break;
```
To preserve the original logic? If we find a POW solution, do not try other solutions.

Reply:
As I can tell, check_tromp_solution does not change the original logic: 
the original logic breaks the loop if a solution is found
and check_tromp_solution in this case just returns also breaking solution searching.
It looks like the boolean return is needed for tests only

Reply:
Yes, seems you are right. The logic is fine.

p.s.  It was "small display issue" (I mixed up the scopes somehow and what it breaks).